sdk/storage/AzureStorageBlob/Source/NetworkingPolicy.swift

```diff
@@ -0,0 +1,95 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import Foundation
+
+/// Describes the type of network connection
+public struct NetworkType: OptionSet {
+    public typealias RawValue = Int
+    public let rawValue: Int
+
+    public static let wifiOrEthernet = NetworkType(rawValue: 1 << 0)
+    public static let cellular = NetworkType(rawValue: 1 << 1)
+
+    // MARK: Initializers
+
+    public init(rawValue: Self.RawValue) {
+        self.rawValue = rawValue
+    }
+}
+
+internal struct NetworkTypeInternal: OptionSet {
```

Comment:
I might just call this `NetworkState` instead, since it tracks the current state of the network.

Also, I understand your comment about making them both OptionSets to be consistent, but it doesn't accurately model reality that way. The network can only be in one state at a time, but the customer can express a desire for multiple network states. So unless it makes the implementation extremely ugly, I would suggest we change this one back to a simple Enum type.

Reply:
Done